src/app/AttributeAccessInterface.h

```diff
@@ -101,6 +102,29 @@ class AttributeValueEncoder : protected TagBoundEncoder
     const FabricIndex mAccessingFabricIndex;
 };
 
+class AttributeValueDecoder
+{
+public:
+    AttributeValueDecoder(TLV::TLVReader * aReader) : mReader(aReader) {}
```

Comment:
The only consumer always has a non-null TLVReader, right?  We should take it by reference, have mReader be a reference, and remove the null-check in Decode.